FirebaseRemoteConfig/SwiftNew/RemoteConfigValue.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import Foundation
+
+@objc(FIRRemoteConfigValue)
+public class RemoteConfigValue: NSObject, NSCopying {
+  /// Data backing the config value.
+  @objc public let dataValue: Data
+
+  /// Identifies the source of the fetched value. Only for Firebase internal use.
+  @objc public let source: RemoteConfigSource
+
+  /// Designated initializer. Only for Firebase internal use.
+  @objc public init(data: Data, source: RemoteConfigSource) {
+    dataValue = data
+    self.source = source
+  }
+
+  /// Gets the value as a string.
+  @objc public var stringValue: String {
+    if let string = String(data: dataValue, encoding: .utf8) {
+      return string
+    }
+    return "" // Return empty string if data is not valid UTF-8
+  }
+
+  /// Gets the value as a number value.
+  @objc public var numberValue: NSNumber {
+    return NSNumber(value: Double(stringValue) ?? 0)
+  }
+
+  /// Gets the value as a boolean.
+  @objc public var boolValue: Bool {
+    return (stringValue as NSString).boolValue
+  }
+
+  /// Gets a foundation object (NSDictionary / NSArray) by parsing the value as JSON.
+  @objc(JSONValue) public var jsonValue: Any? {
+    guard !dataValue.isEmpty else {
+      return nil
+    }
```

Comment:
I think this is a subtle logic change from the ObjC implementation (whether it makes a difference or not, I'm unsure).

The `dataValue` property was nullable in ObjC so ObjC did a nil check:
```objc
- (id)JSONValue {
  NSError *error;
  if (!_data) {
    return nil;
  }
// ...
```
This is doing an an empty check whereas an empty Data object (`[[NSData alloc] init]`) would have preceded to the JSON serialization in the ObjC implementation. The end result might be the same if serialization of an empty data object fails and nil is returned.

Reply:
Updated so the initializer ensures an empty, non-nil Data